src/components/SignIn/SignInForm.tsx

```diff
@@ -0,0 +1,64 @@
+import { Button, Container, FormLabel, Input } from "@chakra-ui/react";
+import React from "react";
+import { useForm } from "react-hook-form";
+import { ErrorMessage } from "@hookform/error-message";
+import useSignIn from "./useSignIn";
+
+interface FormData {
+  email: string;
+  password: string;
+}
+
+const SignInForm = () => {
+  const { isLoading, handleSignInSubmit } = useSignIn();
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm<FormData>();
+  const formSubmit = ({ email, password }: FormData) => {
+    handleSignInSubmit(email, password);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(formSubmit)}>
+      <FormLabel fontWeight="600">Email</FormLabel>
+      <Input
+        placeholder="Enter email"
+        role="email"
+        type="email"
+        {...register("email", {
+          required: "Username is required",
+        })}
+      />
+      <Container color="red">
+        <ErrorMessage errors={errors} name="email" render={({ message }) => <p>{message}</p>} />
+      </Container>
+      <FormLabel marginTop="24px" fontWeight="600">
+        Password
+      </FormLabel>
+      <Input
+        placeholder="Enter password"
+        role="password"
+        type="password"
+        {...register("password", {
+          required: "Password is required",
+        })}
+      />
+      <Container color="red">
+        <ErrorMessage errors={errors} name="password" render={({ message }) => <p>{message}</p>} />
```

Comment:
The function inside of these 2 renders in ErrorMessage can be extracted to a standalone one. Makes this code DRYer.

Reply:
done